### ISSUE TITLE
SSD DNN - PriorBox layers are not working

### ISSUE THREAD

**Issue by bnbhehe** [opencv/opencv#18950]
##### System information (version)

- OpenCV => : 4.4.0.46
- Operating System / Platform => Ubuntu 18.04 64 Bit

##### Steps to reproduce

[config.zip](https://github.com/opencv/opencv/files/5608843/config.zip) <- pb.txt  Text Graph file.

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc
   

Hello, I am having an issue with an exported graph for SSD Detection of vehicles. The model is SSD with custom FPN modulation, predicting 4 classes & having 5 box outputs.

## A bit of context about my procedure 

I am actually doing a pipeline where I have trained on **Pytorch** and follow these steps:

1) Export to ONNX 
2) Convert [ONNX to Keras](https://github.com/waltYeh/onnx2keras)
3) Freeze Keras model to protobuf
4) Create SSD Text graph with a customized version of OpenCV DNN samples (Since I am not using Object Detection API)

My above procedure works, it's a tested recipe and has worked for **RetinaFace** (which is a face detector with 2 classes, 1 aspect ratio, so **simpler prior box hyperparameters**). 

## My current problem

By using the attached `config.pbtxt` file, I am running into the following result
![Selection_727](https://user-images.githubusercontent.com/7895915/100466718-a8030380-30d1-11eb-8dd0-fe177c1bd730.png)

What technically looks to be happening, is that the PriorBox layers seem to be activated with pretty much everything and tile boxes everywhere. The attached video proceeds with showing a vehicle and if you see the attached image below, behind all these is the correct box for it:
![image](https://user-images.githubusercontent.com/7895915/100466853-e0a2dd00-30d1-11eb-9b50-91acbf215719.png)

So I am wondering what's wrong here and why is this happening. As I said my procedure is not the standard TF Object Detection API approach but In the following section you will see that this is not the problem here exactly.

### Check for model weights

Weights are for sure loaded, and actually the model has also been tested on a blank (from black to white) image where the pytorch version is not giving any box predictions.

### Set NMS threshold in DetectionOutputLayer node to a small value. 
This seems to diminish some of the boxes eventually, but the problem persists that boxes are tiled to a black/white canvas. So the PriorBox layers are still working incorrectly.

### Compare logits from output nodes 

By loading only the `.pb` file and treat it as a feed forward network, I was comparing pytorch numpy & OpenCV numpy outputs with ```np.testing.assert_allclose``` assertion. My outputs are the ones specified here:

```
conf_outputs': ['Conv54/Conv2D', 'Conv58/Conv2D', 'Conv62/Conv2D', 'Conv66/Conv2D', 'Conv69/Conv2D'] # ClassPredictor layers
bbox_outputs': ['Conv56/Conv2D', 'Conv60/Conv2D', 'Conv64/Conv2D', 'Conv68/Conv2D', 'Conv70/Conv2D'] # BoxEncodingPredictor layers
```

 There is a tolerance mismatch of 5 digits, but that's insignificant. So the logits are the same. **PS** The graph doesn't contain BiasAdd layers, even when i do net.getLayerNames(). But that's the case also for any model i think because my RetinaFace detector is the same collection of layers. 

### Use Pytorch PriorBox implementation and OpenCV Logits.

Following the above approach of outputing logits, I feed the logits into the Pytorch PriorBox module, and got pretty close results (or at least interpretable), as seen below:
![pytorch_vs_opencv](https://user-images.githubusercontent.com/7895915/100468128-16e15c00-30d4-11eb-880a-32b23b02884b.png)

Red is **Pytorch**, Green is **OpenCV**. I understand that these mismatches might be insignificant, but the gist here is that the model itself doesn't  predict garbage boxes as seen in my first image, which are tiled all over the place.

### About reproducability

I can so far provide the config and if is possible to be inspected and tell me if there is something wrong. If you also need the weights I can provide a model.

Thank you in advance.

**Comment by dkurt** [comment]
Please take a look at one of the works with symbolic PriorBox layer wrapping from PyTorch: https://github.com/amdegroot/ssd.pytorch/pull/462. Maybe it can help to find a problem
